src/createReducedStream.ts

```diff
@@ -0,0 +1,54 @@
+import { BehaviorSubject, Observable, Subscription, Subject } from 'rxjs';
+import { ActionStream } from './types/helpers';
+import { RegisteredReducer, combineReducers } from './reducer';
+import { defaultErrorSubject } from './internal/defaultErrorSubject';
+import { markName } from './internal/markers';
+import { tag } from 'rxjs-spy/operators';
+
+export class ReducedStream<State> extends BehaviorSubject<State> {
+  subscription: Subscription;
+
+  constructor(initialState: State, reducedState$: Observable<State>) {
+    super(initialState);
+
+    this.subscription = reducedState$.subscribe(this);
```

Comment:
**This is indefinite subscription of the stream**

I believe this makes sense, as the global streams should always be active. The cost is, from what I can tell, only the size of `initialState`. We are still free to run initialization logic whenever we want.

Reply:
Does calling `.complete()` on the behavior tear down the subscription?

Reply:
I think that would be the best. I'm revisiting this implementation today.